packages/bridge-ui-v2/src/components/Bridge/Actions.svelte

```diff
@@ -49,6 +49,7 @@
 
   //TODO: this should probably be checked somewhere else?
```

Comment:
```
  function getSpenderAddress(spenderAddresses: any, tokenType: string) {
    const addressMap: Record<string, string> = {
      [TokenType.ERC1155]: 'erc1155VaultAddress',
      [TokenType.ERC721]: 'erc721VaultAddress',
    };

    return spenderAddresses?.[addressMap[tokenType]] || '';
  }

  //TODO: this should probably be checked somewhere else?
  export async function checkTokensApproved() {
    if ($selectedToken?.type === TokenType.ERC721 || $selectedToken?.type === TokenType.ERC1155) {
      if ($account?.address && $network?.id && $destNetwork?.id) {
        const currentChainId = $network?.id;
        const destinationChainId = $destNetwork?.id;
        const token = $selectedToken as NFT;

        const result = await checkOwnershipOfNFT(token as NFT, $account?.address, currentChainId);

        if (!result.every((item) => item.isOwner === true)) {
          return false;
        }

        const wallet = await getConnectedWallet();
        const spenderAddresses = routingContractsMap[currentChainId][destinationChainId];

        const args: RequireApprovalArgs = {
          tokenAddress: token.addresses[currentChainId],
          owner: wallet.account.address,
          spenderAddress: getSpenderAddress(spenderAddresses, token.type),
          tokenId: BigInt(token.tokenId),
          chainId: currentChainId,
        };

        if (token.type === TokenType.ERC1155) {
          try {
            const bridge = bridges[token.type] as ERC1155Bridge;

            // Let's check if the vault is approved for all ERC1155
            const result = await bridge.isApprovedForAll(args);
            allTokensApproved = result;
          } catch (error) {
            console.error('isApprovedForAll error');
          }
        } else if (token.type === TokenType.ERC721) {
          const bridge = bridges[token.type] as ERC721Bridge;

          // Let's check if the vault is approved for all ERC721
          try {
            const requiresApproval = await bridge.requiresApproval(args);
            allTokensApproved = !requiresApproval;
          } catch (error) {
            console.error('isApprovedForAll error');
          }
        }
      }
    }
  }
```
I think this will be better. Avoid repeating code